File: SmartLink/Source/SmartLinkClient/NRC_ServiceGenericClasses/DBLayerClasses/ADO_AccessLayer.vb
Namespace Miscellaneous

    Public Class ADO_AccessLayer

        Public Shared Function DBCommandExecuteScalar(ByVal DBCommandObject As Data.Common.DbCommand) As Object
            Return DBCommandExecuteScalar(DBCommandObject, 3, 30)
        End Function

        Public Shared Function DBCommandExecuteScalar(ByVal DBCommandObject As Data.Common.DbCommand, ByVal DBRetryCount As Integer, ByVal DBRetryInterval As Integer) As Object
            Dim iCounter As Integer

            Do While iCounter < DBRetryCount Or DBRetryCount < 0
                Try
                    Return DBCommandObject.ExecuteScalar()
                Catch ex As Exception
                    If iCounter < DBRetryCount Then
                        iCounter += 1
                        Exit Try
                    End If
                    ex.Data("ADO_AccessLayer::DBExecuteScalar::SQL") = DBCommandObject.CommandText
                    Throw
                End Try
                If DBRetryInterval > 0 Then
                    System.Threading.Thread.Sleep(New TimeSpan(0, 0, DBRetryInterval))
                End If
            Loop

            Return Nothing
        End Function

        Public Shared Function DBCommandExecuteReader(ByVal DBCommandObject As Data.Common.DbCommand) As Data.Common.DbDataReader
            Return DBCommandExecuteReader(DBCommandObject, 3, 30)
        End Function

        Public Shared Function DBCommandExecuteReader(ByVal DBCommandObject As Data.Common.DbCommand, ByVal DBRetryCount As Integer, ByVal DBRetryInterval As Integer) As Data.Common.DbDataReader

            Dim iCounter As Integer = 0

            Do While iCounter < DBRetryCount Or DBRetryCount < 0
                Try
                    Return DBCommandObject.ExecuteReader()
                Catch ex As Exception
                    If iCounter < DBRetryCount Then
                        iCounter += 1
                        Exit Try
                    End If
                    ex.Data("ADO_AccessLayer::DBCommandExecuteReader::SQL") = DBCommandObject.CommandText
                    Throw
                End Try
                If DBRetryInterval > 0 Then
                    System.Threading.Thread.Sleep(New TimeSpan(0, 0, DBRetryInterval))
                End If
            Loop

            Return Nothing
        End Function

        Public Shared Function DBCommandExecuteNonQuery(ByVal DBCommandObject As Data.Common.DbCommand) As Integer
            Return DBCommandExecuteNonQuery(DBCommandObject, 3, 30)
        End Function

        Public Shared Function DBCommandExecuteNonQuery(ByVal DBCommandObject As Data.Common.DbCommand, ByVal DBRetryCount As Integer, ByVal DBRetryInterval As Integer) As Integer
            Dim iCounter As Integer = 0

            Do While iCounter < DBRetryCount Or DBRetryCount < 0
                Try
                    Return DBCommandObject.ExecuteNonQuery()
                Catch ex As Exception
                    If iCounter < DBRetryCount Then
                        iCounter += 1
                        Exit Try
                    End If
                    ex.Data("ADO_AccessLayer::DBCommandExecuteNonQuery::SQL") = DBCommandObject.CommandText
                    Throw
                End Try
                If DBRetryInterval > 0 Then
                    System.Threading.Thread.Sleep(New TimeSpan(0, 0, DBRetryInterval))
                End If
            Loop

            Return 0

        End Function

    End Class

End Namespace

File: SmartLink/Source/SmartLinkClient/NRC_ServiceGenericClasses/DBLayerClasses/ADO_AccessLayer.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Diagnostics;
namespace NRC.Miscellaneous
{

	public class ADO_AccessLayer
	{

		public static object DBCommandExecuteScalar(DbCommand DBCommandObject)
		{
			return DBCommandExecuteScalar(DBCommandObject, 3, 30);
		}

		public static object DBCommandExecuteScalar(DbCommand DBCommandObject, int DBRetryCount, int DBRetryInterval)
		{
			int iCounter = 0;

			while (iCounter < DBRetryCount) 
            {
				try 
                {
                    return DBCommandObject.ExecuteScalar();
				} 
                catch (Exception ex) 
                {
                    iCounter += 1;
                    if (iCounter >= DBRetryCount)
                    {
                        ex.Data["ADO_AccessLayer::DBExecuteScalar::SQL"] = DBCommandObject.CommandText;
                        throw;
                    }
				}
				
                if (DBRetryInterval > 0) 
                {
					System.Threading.Thread.Sleep(new TimeSpan(0, 0, DBRetryInterval));
				}
			}
			return null;
		}

		public static DbDataReader DBCommandExecuteReader(DbCommand DBCommandObject)
		{
			return DBCommandExecuteReader(DBCommandObject, 3, 30);
		}

		public static DbDataReader DBCommandExecuteReader(DbCommand DBCommandObject, int DBRetryCount, int DBRetryInterval)
		{

			int iCounter = 0;

			while (iCounter < DBRetryCount) {
				try {
					return DBCommandObject.ExecuteReader();
				} catch (Exception ex) {
                    iCounter += 1;
                    if (iCounter >= DBRetryCount)
                    {
                        ex.Data["ADO_AccessLayer::DBCommandExecuteReader::SQL"] = DBCommandObject.CommandText;
                        throw;
                    }
				}
				if (DBRetryInterval > 0) {
					System.Threading.Thread.Sleep(new TimeSpan(0, 0, DBRetryInterval));
				}
			}

			return null;
		}

		public static int DBCommandExecuteNonQuery(DbCommand DBCommandObject)
		{
			return DBCommandExecuteNonQuery(DBCommandObject, 3, 30);
		}

		public static int DBCommandExecuteNonQuery(DbCommand DBCommandObject, int DBRetryCount, int DBRetryInterval)
		{
			int iCounter = 0;

			while (iCounter < DBRetryCount || DBRetryCount < 0) {
				try {
					return DBCommandObject.ExecuteNonQuery();
				} catch (Exception ex) {
                    iCounter += 1;
                    if (iCounter >= DBRetryCount)
                    {
                        ex.Data["ADO_AccessLayer::DBCommandExecuteNonQuery::SQL"] = DBCommandObject.CommandText;
                        throw;
                    }
				}
				if (DBRetryInterval > 0) {
					System.Threading.Thread.Sleep(new TimeSpan(0, 0, DBRetryInterval));
				}
			}

			return 0;

		}

	}

}
